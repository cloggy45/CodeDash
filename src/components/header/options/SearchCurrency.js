import {
    connect
} from "react-redux";

import {
    setSelectedCoin
} from "../../../actions/selected";
import {
    fetchTickers,
} from "../../../actions/api";

import {getTickers} from "../../../reducers/rootReducer";

import {withStyles} from '@material-ui/core/styles';

import Options, {styles} from './SearchCurrencyView.js';

const mapStateToProps = (state) => {
    return {
        options: getTickers(state)
    };
};

const mapDispatchToProps = dispatch => {
    return {
        setSelectedCoin: (symbol, id) => {
            dispatch(setSelectedCoin(symbol, id));
        },
        getTickers: () => {
            dispatch(fetchTickers());
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Options));