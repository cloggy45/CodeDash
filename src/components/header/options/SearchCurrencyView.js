import React from "react";
import Select from "react-select";
import "react-select/dist/react-select.css";

export const styles = {
  flex : {
    flexGrow: 1
  }
}

class Options extends React.Component {
  state = {
    selectedOption: "BTC"
  };

  componentDidMount() {
    this.props.getTickers();
    this.props.setOption("BTC");
  }

  handleChange = selectedOption => {
    this.setState({ selectedOption: selectedOption.value }, () => {
      this.props.setOption(this.state.selectedOption);
    });
  };

  formatOptions = (options = []) => {
    return options.map(option => {
      return { value: option.symbol, label: option.name };
    });
  };

  render() {
    const {options, classes} = this.props;
    const { selectedOption } = this.state;

    if(options === null) {
      return <p>Loading....</p>
    } else {
      return (
        <Select
          className={classes.flex}
          name="form-field-name"
          value={selectedOption}
          onChange={this.handleChange}
          options={this.formatOptions(options)}
          clearable={false}
        />
      );
    }
  }
}

export default Options;
