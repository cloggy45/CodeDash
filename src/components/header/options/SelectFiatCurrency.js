import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';

import { setSelectedCurrency } from '../../../actions/selected';
import { getSelectedCurrency} from '../../../reducers/rootReducer';


import SelectCurrency, { styles } from './SelectFiatCurrencyView';

const mapStateToProps = store => ({
    selectedCurrency: getSelectedCurrency(store)
});

const mapDispatchToProps = dispatch => ({
    setSelectedCurrency: (currency) => dispatch(setSelectedCurrency(currency))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(withStyles(styles)(SelectCurrency));
