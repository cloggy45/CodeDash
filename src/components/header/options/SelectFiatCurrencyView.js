import React from 'react';

import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import PropTypes from 'prop-types';

export const styles = {
    root: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    formControl: {
        margin: '1em',
        minWidth: 120,
    },
    selectEmpty: {
        marginTop: '1em',
    },
};

class SelectCurrency extends React.Component {
    handleSetSelectedCurrencyChange = event => {
        const newSelectedFiatCurrency = event.target.value;
        const { selectedCrypto } = this.props;

        this.props.setSelectedCurrency(newSelectedFiatCurrency);
        this.props.fetchCoinPriceInfo(selectedCrypto, newSelectedFiatCurrency);
    };

    render() {
        const { classes, selectedFiat } = this.props;
        return (
            <FormControl className={classes.formControl}>
                <InputLabel htmlFor="currency">Fiat Currency</InputLabel>
                <Select
                    value={selectedFiat}
                    onChange={event =>
                        this.handleSetSelectedCurrencyChange(event)
                    }
                    inputProps={{
                        name: 'currency',
                        id: 'select-currency',
                    }}
                >
                    <MenuItem value={'USD'}>USD</MenuItem>
                    <MenuItem value={'EUR'}>EUR</MenuItem>
                </Select>
            </FormControl>
        );
    }
}

SelectCurrency.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default SelectCurrency;
