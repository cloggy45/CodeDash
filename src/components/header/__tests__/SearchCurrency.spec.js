import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Options, styles  } from '../options/SearchCurrencyView';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchCurrency Component', () => {
    let getTickers, setOptions;

    beforeEach(() => {
        getTickers = jest.fn();
        setOptions = jest.fn();
    });

    it('Should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Options  classes={styles} getTickers={getTickers} setOption={setOptions}/>, div);
    });

    it('Should call getTickers initially when loading', () => {
        const component = shallow(<Options getTickers={getTickers}/>);
        expect(getTickers).toHaveBeenCalledTimes(1);
    });

    it('Should render loading if there are no options', () => {
        const tree= renderer.create(<Options options={null} />).toJSON();
        expect(tree).toMatchSnapshot();

    });
});