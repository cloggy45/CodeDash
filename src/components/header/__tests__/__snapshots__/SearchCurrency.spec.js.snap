// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCurrency Component Should render loading if there are no Selects 1`] = `
<div
  className="Select undefined undefined has-value is-searchable Select--single"
  style={undefined}
>
  <input
    disabled={false}
    name="Search Currency"
    type="hidden"
    value=""
  />
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-3--value"
    >
      <div
        className="Select-value"
        style={undefined}
        title={undefined}
      >
        <span
          aria-selected="true"
          className="Select-value-label"
          id="react-select-3--value-item"
          role="option"
        />
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-3--value"
          aria-describedby={undefined}
          aria-expanded="false"
          aria-haspopup="false"
          aria-label={undefined}
          aria-labelledby={undefined}
          aria-owns=""
          className={undefined}
          id={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          style={
            Object {
              "boxSizing": "content-box",
              "width": "5px",
            }
          }
          tabIndex={undefined}
          value=""
        />
        <div
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "scroll",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
              "whiteSpace": "pre",
            }
          }
        >
          
        </div>
      </div>
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;
